### PR TITLE
ci: add pipelines for build and build-and-release

### PR DESCRIPTION
**Pull request by neelanjan00**
Add CI pipelines for build and build-and-release

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Build
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```
Comment (niladrih) [inline .github/workflows/build.yaml:7]: Pull requests to feature and/or release branches should be testable..
```suggestion
  pull_request:
    types: ['opened', 'edited', 'reopened', 'synchronize']
```

File: .github/workflows/build-and-release.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Build binary
+        uses: houseabsolute/actions-rust-cross@v0
+        with:
+          working-directory: echidna-lb
+          command: ${{ matrix.platform.command }}
+          target: ${{ matrix.platform.target }}
+          args: "--locked --release"
```
Comment (niladrih) [inline .github/workflows/build-and-release.yaml:47]: Should this binary be statically linked?

File: .github/workflows/build-and-release.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            target: x86_64-pc-windows-msvc
+
+          - os-name: macOS-x86_64
+            runs-on: macOS-latest
+            target: x86_64-apple-darwin
+
+    runs-on: ${{ matrix.platform.runs-on }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (niladrih) [inline .github/workflows/build-and-release.yaml]: Should use the same versions for workflows across the board?

**Comment by niladrih** [comment]
Maybe...?
`chore: Add CI pipelines for build and build-and-release` --> `ci: add pipelines for build and build-and-release`

File: .github/workflows/build-and-release.yaml (hunk 2 above)
Comment (neelanjan00) [reply .github/workflows/build-and-release.yaml:47]: Yes, better not to keep an OS dependency. Can you please suggest how to add it?

File: .github/workflows/build-and-release.yaml (hunk 3 above)
Comment (neelanjan00) [reply .github/workflows/build-and-release.yaml]: Updated to v4.

**Comment by neelanjan00** [comment]
> Maybe...? `chore: Add CI pipelines for build and build-and-release` --> `ci: add pipelines for build and build-and-release`

Updated.

File: .github/workflows/build-and-release.yaml (hunk 2 above)
Comment (niladrih) [reply .github/workflows/build-and-release.yaml:47]: Maybe add the arch as a matrix field, and try to build using musl?
```yaml
          - os-name: Linux-x86_64
            runs-on: ubuntu-20.04
            target: x86_64-unknown-linux-musl
            arch: x86_64

          - os-name: Linux-aarch64
            runs-on: ubuntu-20.04
            target: aarch64-unknown-linux-musl
            arch: aarch64
```

```yaml
          args: "--locked --release --target=${{ matrix.arch }}-unknown-linux-musl
```
Mac should work the same way I think.
You'd have to take a stab at riscv64 on your own..

Comment (neelanjan00) [reply .github/workflows/build-and-release.yaml:47]: The action uses **cross** for cross-compilation. Will this arg work with `cross build` command? Ref: https://github.com/houseabsolute/actions-rust-cross

Comment (niladrih) [reply .github/workflows/build-and-release.yaml:47]: Hmmm... I don't know. Let's maybe do the static linking bit separately in a different PR..

**Review by niladrih — APPROVED** [review]

**Pull request merged**